### PR TITLE
Order the getContactIDsFromComponent query results

### PR DESCRIPTION
**Pull request by ray-wright**
Overview
----------------------------------------
With Record generated letters set to "Multiple activities (one per contact)" when you print/merge documents directly on a case, they don't file on the correct contact. Full issue here: https://lab.civicrm.org/dev/core/issues/893

Before
----------------------------------------
When selecting the following cases/contacts in this order (custom sort order on the results page)

Case Number 103, Contact 560 (Linda)
Case Number 112, Contact 579 (Victor)
Case Number 116, Contact 622 (Henry)
Activities are filed in this order

560 (Linda) - filed correctly on case 103
622 (Henry) - filed INCORRECTLY on case 112 (that case belongs to Victor)
579 (Victor) - filed INCORRECTLY on case 116 (that case belongs to Henry)
Rather than the list of contact IDs respecting the sort order, they are in case order.

After
----------------------------------------
When selecting the following cases/contacts in this order (custom sort order on the results page)

Case Number 103, Contact 560
Case Number 112, Contact 579
Case Number 116, Contact 622
Activities are filed in this order

560 - filed correctly on case 103
579 - filed correctly on case 112
622 - filed correctly on case 116

Technical Details
----------------------------------------
_If the PR introduces noteworthy technical changes, please describe them here. Provide code snippets if necessary_

Comments
----------------------------------------
There's an original PR (I closed it) with some discussion here https://github.com/civicrm/civicrm-core/pull/15364. I created it a bit ago and needed to remake it to avoid file conlficts.

### CODE REVIEW

File: CRM/Core/Form/Task.php [hunk 1]
```diff
@@ -177,11 +177,46 @@ public static function preProcessCommon(&$form) {
    * For example, for cases we need to override this function as the table name is civicrm_case_contact
    */
   public function setContactIDs() {
-    $this->_contactIds = CRM_Core_DAO::getContactIDsFromComponent($this->_entityIds,
+    $this->_contactIds = &getContactIDsFromComponent($this->_entityIds,
```
Comment (seamuslee001) [inline CRM/Core/Form/Task.php]: is there a reason why we need the & here?

Comment (ray-wright) [reply CRM/Core/Form/Task.php]: Apologies, it was in the older version of the file - so no.

**Comment by ray-wright** [comment]
@demeritcowboy Here is the new PR - with no more file conflicts.

File: CRM/Core/Form/Task.php [hunk 2]
```diff
@@ -177,11 +177,46 @@ public static function preProcessCommon(&$form) {
    * For example, for cases we need to override this function as the table name is civicrm_case_contact
    */
   public function setContactIDs() {
-    $this->_contactIds = CRM_Core_DAO::getContactIDsFromComponent($this->_entityIds,
+    $this->_contactIds = getContactIDsFromComponent($this->_entityIds,
```
Comment (seamuslee001) [inline CRM/Core/Form/Task.php]: won't it need to be `$this->`

**Comment by demeritcowboy** [comment]
@ray-wright I've put up https://github.com/civicrm/civicrm-core/pull/15626 as an alternate. Using the github UI it's going to be hard for you to update this, so I'll just ask that on the new PR, when it finishes running tests, there'll be link in the middle of the page that says "test sites", and if you can log in there and test it and confirm the filing is working.

**Comment by seamuslee001** [comment]
closing in favour of #15626

**Pull request closed without merging**